### PR TITLE
refactor: partial support for `InvokeMethod2` in `String.flix`

### LINKED ISSUE #7881: Refactor `String.flix` to use `InvokeMethod` etc.
Refactor `String.flix` to use our new Java facilities:

For example, we have:

```
    pub def codePointAt(i: Int32, s: String): Int32 =
        import java.lang.String.codePointAt(Int32): Int32 \ {};
        codePointAt(s, i)
```

which should become:

```
pub def codePointAt(i: Int32, s: String): Int32 = unsafe s¤codePointAt(i)
```

Note: Style-wise, I suggest that if possible we put the whole function on one line.

Of course if the function does more than a single call, we should keep the current formatting.

The `unsafe` keyword here, placed in front, casts the inner expression to pure. This is necessary since the type rule says that every Java method has the IO effect, but we _know_ that e.g. `codePointAt` is a pure function.

### PR DESCRIPTION
**Pull request by chanattan**
Fixes #7881 with current expressiveness of `InvokeMethod2`.

### CODE REVIEW

File: main/src/library/String.flix [hunk 1]
```diff
@@ -1204,28 +1152,14 @@ mod String {
...
     ///
-    pub def indexOfLeftWithOffset(substr: {substr = String}, offset: {offset = Int32}, s: String): Option[Int32] =
-        import java.lang.String.indexOf(String, Int32): Int32 \ {};
-        if (isEmpty(substr.substr))
-            None
-        else {
-            let ix = indexOf(s, substr.substr, offset.offset);
-            if (ix < 0) None else Some(ix)
-        }
+    pub def indexOfLeftWithOffset(substr: {substr = String}, offset: {offset = Int32}, s: String): Option[Int32] = if (isEmpty(substr.substr)) None else { let ix = unsafe s¤indexOf(substr.substr, offset.offset); if (ix < 0) None else Some(ix) }
```
[the remark this answers is not in the document]
Comment (magnus-madsen) [reply main/src/library/String.flix:1155]: What I mean is: If there is just a single method call, we could keep it on the same line.

[the remark this answers is not in the document]
Comment (chanattan) [reply main/src/library/String.flix:1155]: Yes sorry, I parsed your « if possible » too strongly. Fixing this.

[the remark this answers is not in the document]
Comment (magnus-madsen) [reply main/src/library/String.flix:1155]: Might be quicker to revert, and quickly rewrite again.

**Comment by magnus-madsen** [comment]
I made this into a draft because I don't intend to merge this just yet.

**Comment by magnus-madsen** [comment]
No longer relevant (already fixed).

**Pull request closed without merging**